gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/task/DagProcessingEngineMetrics.java

```diff
@@ -75,6 +75,7 @@ public DagProcessingEngineMetrics() {
     List<Tag<?>> tags = new ArrayList<>();
     tags.add(new Tag<>(MetricTagNames.METRIC_BACKEND_REPRESENTATION, GobblinMetrics.MetricType.COUNTER));
     this.metricContext = Instrumented.getMetricContext(new State(), this.getClass(), tags);
+    registerAllMetrics();
```

Comment:
just call `this(this.metricContext)` to use the overloading this class have

Reply:
I tried doing but cannot reference other constructor if not done in first line of method and cannot use `this.` field within the call to constructor

Reply:
```public DagProcessingEngineMetrics() {
    this(Instrumented.getMetricContext(new State(), DagProcessingEngineMetrics.class, Collections.singleton(new Tag<>(MetricTagNames.METRIC_BACKEND_REPRESENTATION, GobblinMetrics.MetricType.COUNTER))));
  }
  this works